tests/abi/tools/revng-ensure-rft-equivalence/Main.cpp

```diff
@@ -88,83 +117,88 @@ int main(int Argc, char *Argv[]) {
     }
   };
   std::map<model::Identifier, FunctionPair, TransparentComparator> Functions;
-
-  // Gather all the `RawFunctionType`s present in the first model.
-  // Make sure there are no duplicated names.
-  for (model::UpcastableType &LeftType : LeftModel->Model->Types()) {
-    if (auto *Left = llvm::dyn_cast<model::RawFunctionType>(LeftType.get())) {
-      revng_assert(LeftModel->Model->DefaultPrototype().isValid());
-      if (Left->ID() == LeftModel->Model->DefaultPrototype().get()->ID()) {
-        // Skip the default prototype.
-        continue;
-      }
-
-      revng_assert(Left->CustomName() != "",
-                   "This test uses names to differenciate functions, as such "
-                   "having unnamed functions in the model would break it, "
-                   "hence it's not allowed.");
-
-      auto [Iterator, Success] = Functions.try_emplace(Left->CustomName());
-      revng_assert(Success, "Two functions share the name in the left model");
-      Iterator->second.Left = Left;
+  std::unordered_set<std::size_t> FunctionIDLookup;
```

Comment:
I'll take a look.

Note: as I said multiple times, I do not look at the floating commits you link. I never did, I never will, since I don't have guarantees that I'm reviewing things that are actually in the PR, in particular after one week from when you post the comment.
Please tell me the name of the function I need to review.
I really really really never look at the commits you link, so please don't link them and mention the function instead.

Reply:
Unless the file is really big (like `Layout.cpp` or `Conversion.cpp`), it usually means that the entire file is affected (like here).